packages/ember-engines/lib/engine-addon.js

```diff
@@ -343,7 +364,13 @@ function buildEngine(options) {
       }
 
       hostAddons[addon.name] = addon;
-      queue.push.apply(queue, addon.addons);
+
+      queue.push.apply(
+        queue,
+        TARGET_INSTANCE_SYMBOL && addon[TARGET_INSTANCE_SYMBOL]
+          ? addon[TARGET_INSTANCE_SYMBOL].addons
+          : addon.addons
+      );
```

Comment:
I know this is pre-existing, but can you migrate to using spread syntax?
```suggestion
      queue.push(
        TARGET_INSTANCE_SYMBOL && addon[TARGET_INSTANCE_SYMBOL]
          ? ...addon[TARGET_INSTANCE_SYMBOL].addons
          : ...addon.addons
      );
```